File: apt-integration-tests/src/test/java/me/kpavlov/kt/schema/apt/integration/type/StatusSchemaTest.java
package me.kpavlov.kt.schema.apt.integration.type;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;

import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;

/**
 * Verifies the kt-schema-apt processor generates a JSON Schema resource for a Java enum.
 */
class StatusSchemaTest {

    private static final String RESOURCE_PATH =
            "META-INF/kt-schema/schemas/me/kpavlov/kt/schema/apt/integration/type/Status.json";

    @Test
    void shouldGenerateEnumSchemaWithAllValues() throws IOException {
        // language=json
        assertThatJson(readGeneratedSchema()).isEqualTo("""
                {
                  "$id": "me.kpavlov.kt.schema.apt.integration.type.Status",
                  "$schema": "https://json-schema.org/draft/2020-12/schema",
                  "type": "string",
                  "enum": ["ACTIVE", "INACTIVE", "PENDING"],
                  "description": "Current lifecycle status of an entity."
                }
                """);
    }

    private static String readGeneratedSchema() throws IOException {
        try (InputStream input = StatusSchemaTest.class.getClassLoader().getResourceAsStream(RESOURCE_PATH)) {
            if (input == null) {
                throw new IllegalStateException("Missing generated schema resource: " + RESOURCE_PATH);
            }
            return new String(input.readAllBytes(), StandardCharsets.UTF_8);
        }
    }
}

File: ksp-integration-tests/src/test/kotlin/me/kpavlov/kt/schema/integration/type/StatusSchemaTest.kt
package me.kpavlov.kt.schema.integration.type

import io.kotest.assertions.json.shouldEqualJson
import kotlin.test.Test

/**
 * Tests for Status schema generation - enum handling.
 */
class StatusSchemaTest {
    @Test
    fun `generates enum schema with all values`() {
        val schema = Status::class.jsonSchemaString

        // language=json
        schema shouldEqualJson
            $$"""
            {
              "$id": "Status",
              "$schema": "https://json-schema.org/draft/2020-12/schema",
              "type": "string",
              "enum": ["ACTIVE", "INACTIVE", "PENDING"],
              "description": "Current lifecycle status of an entity."
            }
            """.trimIndent()
    }
}
